docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,513 @@
+Person.find("{'status': ?1}", Status.Alive).project("name";
+--
+
+== The DAO/Repository option
+
+Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we know life is tough and we've got you covered.
+
+If you want to have Repositories, you can get the exact same convenient methods injected in your Repository by making it
+implement `PanacheMongoRepository`:
+
+[source,java]
+--
+@ApplicationScoped
+public class PersonRepository implements PanacheMongoRepository<Person> {
+
+   // put your custom logic here as instance methods
+
+   public Person findByName(String name){
+     return find("name", name).firstResult();
+   }
+
+   public List<Person> findAlive(){
+     return list("status", Status.Alive);
+   }
+
+   public void deleteStefs(){
+     delete("name", "Stef");
+  }
+}
+--
+
+Absolutely all the operations that are defined on `PanacheMongoEntityBase` are available on your DAO, so using it
+is exactly the same except you need to inject it:
+
+[source,java]
+--
+@Inject
+PersonRepository personRepository;
+
+@GET
+public long count(){
+  return personRepository.count();
+}
+--
+
+So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
+subclasses of `PanacheMongoEntity` in order to get the ID and public fields working, but you can even skip that and
+go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+
+== Transactions
+
+MongoDB offers ACID transactions since version 4.0. MongoDB with Panache doesn't provides support for them.
+
+== Custom IDs
+
+IDs are often a touchy subject, in MongoDB they are usually auto-generated by the database with an `ObjectId`. 
+In MongoDB with Panache the ID are defined by a field name `id` of the `org.bson.types.ObjectId` type,
+but if you want ot customize them, once again we have you covered.
+
+You can specify your own ID strategy by extending `PanacheMongoEntityBase` instead of `PanacheMongoEntity`. Then
+you just declare whatever ID you want as a public field by annotating it by @BsonId:
+
+[source,java]
+--
+@MongoEntity
+public class Person extends PanacheMongoEntityBase {
+
+  @BsonId
+  public Integer myId;
+
+ ...
+}
+--
+
+If you're using repositories, then you will want to extend `PanacheMongoRepositoryBase` instead of `PanacheMongoRepository`
+and specify your ID type as an extra type parameter:
+
+[source,java]
+--
+@ApplicationScoped
+public class PersonRepository implements PanacheMongoRepositoryBase<Person,Integer> {
+
+ ...
+}
+--
+
+[NOTE]
+====
+When using `ObjectId`, MongoDB will automatically provide a value for you, but if you use a custom field type,
+you need to provides the value by yourself.
+====
+
+`ObjectId` can be difficult to use if you want to expose its value in your rest service.
+So we create JsonB providers to serialize/deserialize them as String.
```

Comment:
What if you use Jackson?

Reply:
Yasson serialize/deserialize it as a complex json object.
Jackson throws an exception! I can provide serializer/deserializer to Jackson also.

Reply:
@gsmet, I add serializer/deserializer as String for Jackson also.